src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -0,0 +1,575 @@
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+end
+
+--[=[
+	Used to track player events (example: player killed an enemy, player completed a mission, etc.)
+	
+	Examples
+	```lua
+	AnalyticsService:LogPlayerEvent({
+		userId = player.UserId,
+		event = "Player:KilledEnemy",
+		value = 1 -- Killed 1 enemy
+	})
+	AnalyticsService:LogPlayerEvent({
+		userId = player.UserId,
+		event = "Player:CompletedMission",
+		value = 1 -- Completed 1 mission
+	})
+	AnalyticsService:LogPlayerEvent({
+		userId = player.UserId,
+		event = "Player:Death",
+		value = 1
+	})
+	```
+]=]
+function AnalyticsService:LogPlayerEvent(data: PlayerEvent)
+	assert(typeof(data) == "table", "AnalyticsService.LogPlayerEvent - data is required")
+	assert(data.userId, "AnalyticsService.LogPlayerEvent - userId is required")
+	assert(data.event, "AnalyticsService.LogPlayerEvent - event is required")
+	assert(
+		data.value == nil or typeof(data.value) == "number",
+		"AnalyticsService.LogPlayerEvent - value must be a number"
+	)
+	
+	GameAnalytics:addDesignEvent(data.userId, {
+		eventId = data.event,
+		value = data.value
+	})
+end
+
+--[=[
+	This function should be called when a successful marketplace purchase is made
+	such as a gamepass or developer product
+	
+	Set `LOG_DEV_PRODUCT_PURCHASES` to false inside AnalyticsService.lua if you prefer to log
+	developer product purchases within MarketplaceService.ProcessReceipt
+	
+	```lua
+	-- Inside MarketplaceService.ProcessReceipt
+	-- before returning Enum.ProductPurchaseDecision.PurchaseGranted
+	AnalyticsService:LogMarketplacePurchase({
+		userId = player.UserId,
+		itemType = "Product",
+		id = 000000000, -- Developer product id
+		cartType = "PromptPurchase",
+	})
+	```
+]=]
+function AnalyticsService:LogMarketplacePurchase(data: MarketplacePurchaseEvent)
+	assert(typeof(data) == "table", "AnalyticsService.LogMarketplacePurchase - data is required")
+	assert(data.userId, "AnalyticsService.LogMarketplacePurchase - userId is required")
+	assert(data.itemType, "AnalyticsService.LogMarketplacePurchase - itemType is required")
+	assert(data.id, "AnalyticsService.LogMarketplacePurchase - id is required")
+	assert(data.cartType, "AnalyticsService.LogMarketplacePurchase - cartType is required")
+	assert(data.userId, "AnalyticsService.LogMarketplacePurchase - userId is required")
+	assert(
+		data.amount == nil or typeof(data.amount) == "number",
+		"AnalyticsService.LogMarketplacePurchase - amount must be a number"
+	)
+	
+	GameAnalytics:addBusinessEvent(data.userId, {
+		itemType = data.itemType,
+		itemId = data.id,
+		amount = data.amount or 1,
+		cartType = data.cartType
+	})
+end
+
+--[=[
+	Shortcut function for LogResourceEvent
+	Used to log in-game currency purchases
+	
+	Example Use:
+	```lua
+	AnalyticsService:LogPurchase({
+		userId = player.UserId,
+		eventType = "Shop",
+		currency = "Coins",
+		itemId = "Red Paintball Gun"
+	})
+	```
+]=]
+function AnalyticsService:LogPurchase(data: PurchaseEvent)
```

Comment:
Add param tag to the moonwave block.
@param data PurchaseEvent -- PurchaseEvent type

Reply:
See earlier note about using promise instead of assert